src/java.desktop/windows/native/libawt/windows/awt_Component.cpp

```diff
@@ -6539,19 +6558,32 @@ static void _GetInsets(void* param)
 {
     JNIEnv *env = (JNIEnv *)JNU_GetEnv(jvm, JNI_VERSION_1_2);
 
-    GetInsetsStruct *gis = (GetInsetsStruct *)param;
+    GetInsetsStruct *gis = static_cast<GetInsetsStruct *>(param);
     jobject self = gis->window;
 
     gis->insets->left = gis->insets->top =
         gis->insets->right = gis->insets->bottom = 0;
 
     PDATA pData;
-    JNI_CHECK_PEER_GOTO(self, ret);
-    AwtComponent *component = (AwtComponent *)pData;
+    if (self == NULL) {
+        env->ExceptionClear();
+        JNU_ThrowNullPointerException(env, "self");
+        delete gis;
+        return;
+    } else {
+        pData = JNI_GET_PDATA(self);
+        if (pData == NULL) {
+            THROW_NULL_PDATA_IF_NOT_DESTROYED(self);
+            env->DeleteGlobalRef(self);
+            delete gis;
+            return;
+        }
+    }
+    AwtComponent *component = (AwtComponent *) pData;
 
     component->GetInsets(gis->insets);
 
-  ret:
+
```

Comment:
I recommend deleting this line, since it does not make sense to have two consecutive blank lines here.